src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs

```diff
@@ -220,6 +223,21 @@ protected void AddMapInternal(string collection, string subCollecction, Indexing
             funcs.Add(map);
         }
 
+        internal void CheckDepthOfStackInOutputMap(IndexDefinition indexMetadata, DocumentDatabase documentDatabase)
+        {
+            var performanceHintConfig = documentDatabase.Configuration.PerformanceHints;
+            if (StackSizeInSelectClause > performanceHintConfig.MaxDepthOfRecursionInLinqSelect)
```

Comment:
I think we want two settings:
* > 20 - we raise an alert
* > 50 - we refuse to allow the index (with the appropriate notice on how to configure that, of course)

Reply:
Refusing to put an index is something that might be a breaking change for someone. This worries me. Consider the case that a new node is added to the cluster - then an index with such problem could not be put.